src/Client/Invoice.php

```diff
@@ -149,4 +150,30 @@ public function getPaymentMethods(string $storeId, string $invoiceId): array
             throw $this->getExceptionByStatusCode($method, $url, $response);
         }
     }
+
+    public function markInvoiceStatus(string $storeId, string $invoiceId, string $markAs): \BTCPayServer\Result\Invoice
```

Comment:
We have constants for the states in `AbstractResult` should we move it elsewhere or document this somewhere so we use them also here? Any thoughts @woutersamaey

Reply:
I looked at `AbstractResult`, but there are no constants there?

Reply:
Sorry, I meant `Result\Invoice`

Reply:
Yeah, checking the inputs would be a nice addition. It should be `Invalid` or `Settled`. Anything else should trigger an `InvalidArgumentException`.

Reply:
Actually, come to think of it, the API would throw an error when an invalid status is used, so maybe it allows for greater flexibility if we don't check in PHP. What if in v2 we get some new statusses for example, then we'd need to change the PHP lib as well.
This method only has 1 function and it is to change the status, so the API has plenty of checks of its own.
I propose we don't add the extra check and let the API handle the error.

Reply:
Ok good point I agree to let it fail on api level then. I don't see the v2 argument though, we likely need to change a lot of places and break bc anyway and go with a new major release.